method.rb

```diff
@@ -0,0 +1,181 @@
+module Enumerable # rubocop:disable Style/ModuleLength
+  def my_each
+    i = 0
+    arr = to_a
+    return to_enum(:my_each) unless block_given?
+
+    while i < arr.size
+      yield arr[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    i = 0
+    arr = to_a
+    return to_enum(:my_each_with_index) unless block_given?
+
+    while i < arr.size
+      yield arr[i], i
+      i += 1
+    end
+
+    self
+  end
+
+  def my_select
+    new_arr = []
+    return to_enum(:my_select) unless block_given?
+
+    my_each do |num|
+      new_arr.push(num) if yield num
+    end
+
+    new_arr
+  end
+
+  def my_all?(arg = nil) # rubocop:disable Style/MethodLength
+    stat = true
+
+    if arg
+      if arg.class == Class
+        my_each do |num|
+          stat = num.is_a?(arg)
+        end
+      else
+        my_each do |num|
+          stat = num.to_s.match?(arg.to_s)
+        end
+      end
+
+    elsif block_given?
+      my_each do |num|
+        stat = false unless yield num
+      end
+    else
+      my_each do |num|
```

Comment:
- [ ] When you don't give a block or argument to the my_all? method it should return true only if all of the enumerable items are 'truthy'.
So an array with a 'falsy' item should return false wit this method. [Read this article if you are not sure which are the falsy and truthy values in ruby.](https://andycroll.com/ruby/truthiness-in-conditionals/)
Fix the code inside your your else conditional to fix this.